### ISSUE TITLE
`jsx-sort-props`: minimum number of props

### ISSUE THREAD

**Issue by Garrett-R** [jsx-eslint/eslint-plugin-react#3922]
Similar to how ESLint's [`sort-keys`](https://eslint.org/docs/latest/rules/sort-keys) has a configuration option of `minKeys`, it would be nice to have a `minProps` option.

**Comment by ljharb** [comment]
Why, exactly?

**Comment by Garrett-R** [comment]
Ah sorry, should have added context.  Two reasons:
1) for components with only a couple props, like say 2, the ordering isn't too important and the rule might be more annoying than helpful to the devs.  
2) devs may want the option to order "semantically" for reasonable number of props, for example:

```tsx
<MyButton
    type="submit"
    color="blue"
    variant="solid"
    onBlur={onBlur}
    onClick={onClick}
/>
```

while if a component has like 10+ props, then I think it should definitely be ordered.

It seems (1) was the motivation for the analogue `sort-keys` being added to ESLint (see convo [here](https://github.com/eslint/eslint/issues/11624)).

**Comment by ljharb** [comment]
I guess it's fine, altho I'm not sure it's a good thing to be tacitly encouraging reduced enforcement.
